tests/unit/test_util_types.py

```diff
@@ -279,3 +278,13 @@ def test_version_compare(self):
         self.assertTrue(Version('4') == Version('4.0.0'))
         self.assertTrue(Version('4.0') == Version('4.0.0.0'))
         self.assertTrue(Version('4.0') > Version('3.9.3'))
+
+        self.assertTrue(Version('4.0-SNAPSHOT') == Version('4.0-SNAPSHOT'))
+        self.assertTrue(Version('4.0.0-SNAPSHOT') == Version('4.0-SNAPSHOT'))
+        self.assertTrue(Version('4.0.0-SNAPSHOT') == Version('4.0.0-SNAPSHOT'))
+        self.assertTrue(Version('4.0.0.build5-SNAPSHOT') == Version('4.0.0.build5-SNAPSHOT'))
+        self.assertTrue(Version('4.1-SNAPSHOT') > Version('4.0-SNAPSHOT'))
+        self.assertTrue(Version('4.0.1-SNAPSHOT') > Version('4.0.0-SNAPSHOT'))
+        self.assertTrue(Version('4.0.0.build6-SNAPSHOT') > Version('4.0.0.build5-SNAPSHOT'))
+        self.assertTrue(Version('4.0-SNAPSHOT2') > Version('4.0-SNAPSHOT1'))
+        self.assertTrue(Version('4.0-SNAPSHOT2') > Version('4.0.0-SNAPSHOT1'))
```

Comment:
Will this work?

```
 self.assertTrue(Version('4.0') > Version('4.0-SNAPSHOT'))
```

Reply:
No, good catch. I didn't think about the meaning of the word "prerelease" :smile: